framework/include/libtorch/userobjects/TorchScriptUserObject.h

```diff
@@ -0,0 +1,53 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
```

Comment:
I'm curious about @loganharbour 's opinion on this. If we ifdef guard the entire file, this object won't be available when libtorch is not enabled. This may create several issues:
- user won't be able to autocomplete using language server
- some civet recipe might not like this

Reply:
In general, I do suggest that we have all of the objects (in a bare bones, error-on-construction state) available even when the capabilities do not exist. Same for the material object.

Reply:
We actually had a design discussion with @loganharbour and @zachmprince when the original implementation went in. The original design was what you are suggesting but based on that discussion we moved to this approach. I am okay changing it back, but in another PR in the future.

Reply:
The problem with fully deactivating is that the documentation generation will error out with "unknown syntax". If we had a fix for this I'd be in favor of fully guarding the objects, because creating an "error on construction" stub can be a _lot_ of tedious work, and leads to files that are chock full of ifdefs.

Reply:
If you are talking about the language server I am not sure about a solution on that front. 
If you are talking about moosedocs, we guard the documentation with:

```
!if function=hasLibtorch()
```

Reply:
That's not a bad approach IMO. Maybe this could be generalized to the new "capability" system with something like
```
!if function=hasCapability("libtorch")
```

Reply:
Well, we also had a design discussion with @loganharbour and @zachmprince and decided that most of the documentation can still exist. What we do is something like this:
```
!if function=hasLibtorch()

!syntax parameters /UserObjects/TorchScriptUserObject

!syntax inputs /UserObjects/TorchScriptUserObject

!syntax children /UserObjects/TorchScriptUserObject

!if-end!

!else
!include libtorch/libtorch_warning.md
```

Which will claim that MOOSE was not compiled with libtorch and says that more content will become available if the executable used for the doco has libtorch support.

Reply:
I am okay doing a sweep if the consensus says that we just want to turn the doco off by default as well. But in a different PR.

Reply:
The problem with the

```
!if function=hasLibtorch()
```

approach is that you have to replicate the exact conditions that you have in the source file to conditionally compile (and those condition can now go out of sync). The only thing the doc system cares about is whether the object exists or not. So why not check _that_?

Reply:
I will think about it, and then move all the existing documentation in one go (in a different PR) if we decide we need another approach. In my opinion the main question is: do we even want a doc entry for these objects or not? In past conversations I was persuaded that we do even when the objects are not compiled. If that still stands I don't see a difference between the two approaches considering that the only way that object is not registered is if libtorch is not there.